### DISCUSSION TITLE
Custom prompt for active note (in Chat)

### DISCUSSION THREAD

**Discussion by Mara-Li** [logancyang/obsidian-copilot#315] [Q&A]
Maybe it's like I don't understand the custom prompt, but I would like to create a prompt on the active note. 
It seems that the variable only include specific note, not the active note (if any)?

I would like a variable to just say "use the active note" instead of changing for the note.

**Comment by logancyang** [comment]
For custom prompt templating, it doesn't matter what the active note is. You reference it the same way as other notes, using its title `{[[Note Title]]}`.

**Reply by Mara-Li** [reply]
So it doesn't exist a template for the active note? Like `{[[active-note]]}` ?

**Reply by logancyang** [reply]
No, I don't see the need for that when you can reference it by title. It's more explicit and less error-prone when you have many notes open at the same time (I often hit the "up" button and find out later the active note is actually still my previous tab, for example).

**Reply by logancyang** [reply]
I think having auto-suggestion for note title, path, and tags when the user types ahead is necessary for better experience, just like typing in a note. It can be the next task.

**Reply by Mara-Li** [reply]
> No, I don't see the need for that when you can reference it by title. It's more explicit and less error-prone when you have many notes open at the same time (I often hit the "up" button and find out later the active note is actually still my previous tab, for example).

The problem is that I need 4/5 clicks to just insert my prompt...

Like I have a prompt that allow me to resume a note in a specific way. But to use the prompt I need:
- Set the active note as contexte
- Return to the tab where active note is
- Use prompt apply command 
- Return to chat

**Reply by logancyang** [reply]
I see what you mean, you want a designated symbol to represent whatever active note is. For now, you can still achieve this just with `{}` and highlight everything in your active note. 

I think that a better and simpler Chat experience is just directly bringing up `[[ title ]]` inside the chat itself, but with typeahead assistance just like inside the note. That was my original vision, so having a dedicated symbol for active note wasn't on my radar.

**Reply by logancyang** [reply]
@Lisandra-dev I don't really follow why you need these steps for a prompt that continues a note. Are you using the "sent note to prompt" button or the "apply custom prompt" command? The command doesn't need context setting, only the button needs it. Maybe I misunderstood.
